src/include/migraphx/op/dot.hpp

```diff
@@ -39,7 +39,7 @@ struct dot
     std::string name() const { return "dot"; }
     shape compute_shape(std::vector<shape> inputs) const
     {
-        check_shapes{inputs, *this, true}.same_type().same_ndims().has(2);
+        check_shapes{inputs, *this, true}.same_ndims().has(2);
```

Comment:
I think they both need to be same for GPU backend atleast. For the Ref we  can do it, but if we are going to allow it, we better test it. Can you explain why you made this change ?

Reply:
It was due to be using add_instruction and replace_instruction as part of the pass. I've used inserts and moves instead to get the desired result for the 1-d and 2d cases now, and it appears that our quantizelinear op was incorrect as only the first two args (data, scale) need be the same time and zero point must be assigned to the desired output type as zero point and output match. 

My recent changes should have fixed this.